test/integration/smapi-commands-test.js

```diff
@@ -0,0 +1,888 @@
+const accountLinkingRequest = require('./fixtures/account-linking-request.json');
+const interactionModel = require('./fixtures/interaction-model.json');
+
+parallel.limit(8);
+
+const processor = new CliCustomizationProcessor();
+const modelIntrospector = new ModelIntrospector();
+const untestedCommands = new Set([...modelIntrospector.getOperations().keys()].map(processor.processOperationName));
+const testedCommands = new Set();
+
+const addCoveredCommand = (args) => {
+    const cmd = args[1];
+    if (testedCommands.has(cmd)) {
+        console.warn(`${cmd} already has been covered!`);
+    }
+    testedCommands.add(cmd);
+    untestedCommands.delete(cmd);
+};
+
+parallel('smapi command test', () => {
+    const cmd = 'ask';
+    const subCmd = 'smapi';
+    let mockSmapiServer;
+    let mockLwaServer;
+    const options = { parse: true,
+        env: {
+            ASK_SMAPI_SERVER_BASE_URL: `http://127.0.0.1:${MockServerPort.SMAPI}`,
+            ASK_LWA_TOKEN_HOST: `http://127.0.0.1:${MockServerPort.LWA}`
+        } };
+    const catalogId = 'someCatalogId';
+    const skillId = 'someSkillId';
+    const productId = 'someProductId';
+    const stage = 'development';
+    const locale = 'en-US';
+    const location = 'US';
+    const uploadId = 'someUploadId';
+    const subscriberId = 'someSubscriberId';
+    const subscriptionId = 'someSubscriptionId';
+    const updateRequestId = 'someUpdateRequestId';
+    const version = '2.0.0';
+    const simulationId = 'someSimulationId';
+    const slotTypeId = 'someSlotTypeId';
+    const slotType = JSON.stringify({
+        slotType: {
+            name: 'string',
+            description: 'string'
+        },
+        vendorId: 'string'
+    });
+    const createSubscriptionRequest = JSON.stringify(
+        {
+            name: 'string',
+            vendorId: 'string',
+            endpoint: {
+                uri: 'string',
+                authorization: {
+                    type: 'string'
+                }
+            }
+        }
+    );
+    const updateSubscriptionRequest = JSON.stringify({
+        name: 'string',
+        endpoint: {
+            uri: 'string',
+            authorization: {
+                type: 'string'
+            }
+        }
+    });
+    const sslCertificatePayload = JSON.stringify({
+        sslCertificate: 'string',
+        regions: {
+            additionalProp1: {
+                sslCertificate: 'string'
+            },
+            additionalProp2: {
+                sslCertificate: 'string'
+            },
+            additionalProp3: {
+                sslCertificate: 'string'
+            }
+        }
+    });
+    const partETags = JSON.stringify([{ eTag: 'someEtag', partNumber: 1 }]);
+    const testersEmails = 'user1@gmail.com,user2@gmail.com';
+
+    before(async () => {
+        mockSmapiServer = await startMockSmapiServer();
+        mockLwaServer = await startMockLwaServer();
+    });
+
```

Comment:
Could you add a test case as a summary test case, to compare the number of existing smapi commands vs. test-covered commands? So that we know if we missed the integ-test for any commands~ I'm not sure if we should mark it failure if number doesn't match, but strictly say we should fail and block the release.

Reply:
@RonWang look at the "after" block

Reply:
nice!